File: easy_25_leap_year_checker/solution.py
"""Reference implementation for the platform's official solution."""

from __future__ import annotations
import sys
from typing import TextIO

class Solution:
    def isLeapYear(self, year):
        """
        :type year: int
        :rtype: bool
        """
        return (year % 4 == 0 and year % 100 != 0) or (year % 400 == 0)

def solve(infile: TextIO = sys.stdin, outfile: TextIO = sys.stdout) -> None:
    year = int(infile.read().strip())
    
    solution = Solution()
    result = solution.isLeapYear(year)
    
    outfile.write(f"{'True' if result else 'False'}\n")

if __name__ == "__main__":
    solve()

File: easy_25_leap_year_checker/solution.cpp
#include <algorithm>
#include <iostream>
#include <iterator>
#include <sstream>
#include <string>
#include <vector>

using namespace std;

class Solution {
public:
    bool isLeapYear(int year) {
        return (year % 4 == 0 && year % 100 != 0) || (year % 400 == 0);
    }
};

void solve(istream& in = cin, ostream& out = cout) {
    int year;
    in >> year;
    
    Solution solution;
    bool result = solution.isLeapYear(year);
    
    out << (result ? "True" : "False") << endl;
}

int main() {
    solve();
    return 0;
}
